lib/connectors/memory.js

```diff
@@ -148,11 +148,13 @@ Memory.prototype.saveToFile = function (result, callback) {
   if (file) {
     if(!self.writeQueue) {
       // Create a queue for writes
+      var cache = self.cache;
+      var ids = self.ids;
       self.writeQueue = async.queue(function (task, cb) {
         // Flush out the models/ids
         var data = JSON.stringify({
-          ids: self.ids,
-          models: self.cache
+          ids: ids,
+          models: cache
```

Comment:
This looks like a possible problem (a race condition?) in the way how we schedule writes via `writeQueue` (`async.queue`), we must pay extra attention before modifying this part of code.

@Amir-61 what is the test case that fails without your change? I would prefer to move this fix to a standalone pull request, so that we can think it through in isolation. There are already too many other changes in this PR.